pandas/tests/test_testing.py

```diff
@@ -215,6 +215,13 @@ def test_multiindex_dtype(self):
                 {'a':[1.0,2.0],'b':[2.1,1.5],'c':['l1','l2']}, index=['a','b'])
         self._assert_not_equal(df1, df2, check_index_type=True)
 
+    def test_empty_dtypes(self):
```

Comment:
``icol`` is used internally, ``iloc`` is a more general form. 

ok, I see what you are trying to test here. But its circular, and not convincing that you are actually checking anything. I appreciate that you are trying to make sure that ``check_dtype`` is 'working'. But that goes back to asserting that ``assert_frame_equal`` is working.